common/src/buffers/types.rs

```diff
@@ -101,39 +76,27 @@ impl DeniablePayload {
         &mut self.denim_chunks
     }
 
-    pub fn garbage(&self) -> &Option<Vec<u8>> {
+    pub fn garbage(&self) -> &Vec<u8> {
         &self.garbage
     }
+}
 
-    pub fn to_bytes(self) -> Result<Vec<Vec<u8>>, DenimBufferError> {
-        let mut encoded_chunks = Vec::new();
-        for chunk in self.denim_chunks {
-            encoded_chunks.push(
-                bincode::encode_to_vec(chunk, config::standard().with_fixed_int_encoding())
-                    .map_err(|_| DenimBufferError::ChunkEncodeError)?,
-            );
-        }
-        if let Some(garbage) = self.garbage {
-            encoded_chunks.push(garbage);
-        }
-        Ok(encoded_chunks)
-    }
-
-    pub fn decode(bytes: Vec<Vec<u8>>) -> Result<Vec<DenimChunk>, DenimBufferError> {
-        let mut denim_chunks = Vec::new();
-        for chunk in bytes {
-            let (mut denim_chunk, _): (DenimChunk, usize) =
-                bincode::decode_from_slice(&chunk, config::standard().with_fixed_int_encoding())
-                    .map_err(|_| DenimBufferError::ChunkDecodeError)?;
+#[derive(Encode, Decode, Builder, Clone)]
+pub struct DenimMessage {
+    pub regular_payload: Vec<u8>,
+    pub deniable_payload: DeniablePayload,
+}
 
-            if denim_chunk.has_garbage_flag() {
-                denim_chunk.remove_garbage_flag();
-                denim_chunks.push(denim_chunk);
-                return Ok(denim_chunks);
-            }
-            denim_chunks.push(denim_chunk);
-        }
+impl DenimMessage {
+    pub fn to_bytes(self) -> Result<Vec<u8>, DenimEncodeError> {
```

Comment:
maybe encode / decode instead of to_bytes / decode
```suggestion
    pub fn encode(self) -> Result<Vec<u8>, DenimEncodeError> {
```